### PR TITLE
Use TelemetryItemExporter in azmon-otel-exporter

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java [hunk 1]
```diff
@@ -200,16 +203,18 @@ public AzureMonitorExporterBuilder clientOptions(ClientOptions clientOptions) {
...
     if (!keyValues.containsKey("InstrumentationKey")) {
       throw logger.logExceptionAsError(
           new IllegalArgumentException("InstrumentationKey not found in connectionString"));
     }
     this.instrumentationKey = keyValues.get("InstrumentationKey");
     String endpoint = keyValues.get("IngestionEndpoint");
-    if (endpoint != null) {
-      this.endpoint(endpoint);
+    if (endpoint == null) {
+      endpoint = "https://dc.services.visualstudio.com";
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java:215]: can we use DefaultEndpoints.INGESTION_ENDPOINT?

File: agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java [hunk 2]
```diff
@@ -119,18 +122,21 @@
     STANDARD_ATTRIBUTE_PREFIXES = Collections.unmodifiableSet(standardAttributesPrefix);
   }
 
-  private final MonitorExporterAsyncClient client;
+  private final TelemetryItemExporter telemetryItemExporter;
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java:125]: nice, TelemetryItemExporter is becoming a reusable exporter now.

**Comment by heyams** [comment]
do we need to add more tests in AzureMonitorExporterBuilderTest with this change? 
It can be a todo: add smoke test for azureMonitorExporterBuilder?

File: agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java:215]: I'm working on sharing ConnectionString parsing in next PR(s), this code will be gone shortly 👍

**Comment by trask** [comment]
> do we need to add more tests in AzureMonitorExporterBuilderTest with this change? It can be a todo: add smoke test for azureMonitorExporterBuilder?

let's chat about what kind of tests you have in mind

**Review by heyams — APPROVED** [review]

**Pull request merged**